lib/sequence_diagram/sequencer.rb

```diff
@@ -0,0 +1,72 @@
+require "ostruct"
+
+module SequenceDiagram
+  class Sequencer
+    def self.sequence
```

Comment:
Sequencer.sequence is a bit redundant. I've adopted Michael Feathers's convention for naming the unit for what it does, and then just uniformly naming the invocation method `run!` which allows task objects to be treated as duck types, it's short, and it doesn't add unnecessary info.